python/ray/data/_internal/logical/operators/from_items_operator.py

```diff
@@ -0,0 +1,40 @@
+from typing import Any, List, TYPE_CHECKING
+
+from ray.data._internal.logical.interfaces import LogicalOperator
+
+if TYPE_CHECKING:
+    import tensorflow as tf
+    import torch
+
+
+class FromItems(LogicalOperator):
+    """Logical operator for `from_items`."""
+
+    def __init__(
+        self,
+        items: List[Any],
+        parallelism: int = -1,
+        op_name: str = "FromItems",
+    ):
+        super().__init__(op_name, [])
+        self._items = items
+        self._parallelism = parallelism
+
+
+class FromTF(FromItems):
```

Comment:
Does this create Arrow blocks or simple blocks? (I assume arrow?) I am wondering if it makes sense to extend the Arrow logical operator if that's the case.

Reply:
Yeah, this creates Arrow blocks I believe:
```
>>> import ray; import tensorflow_datasets as tfds
>>> tf_dataset, _ = tfds.load('cifar10', split=["train", "test"])
>>> dataset = ray.data.from_tf(tf_dataset)
>>> blocks = dataset._plan.execute()
>>> ray.get(blocks._blocks[0])
pyarrow.Table
id: binary
image: extension<arrow.py_extension_type<ArrowTensorType>>
label: int64
```
Since the existing `from_tf` [implementation](https://github.com/ray-project/ray/blob/master/python/ray/data/read_api.py#L1780) relies on `from_items` on a numpy iterator converted to a list, I was thinking this most closely reflects `FromItems`. Thoughts?

Reply:
I think it's fine to refine `FromTF` to extend `FromNumpyRefs`, so it's being more specific.